tests/AtlasSearchTest.php

```diff
@@ -194,9 +203,62 @@ public function testDatabaseBuilderAutocomplete()
         self::assertInstanceOf(LaravelCollection::class, $results);
         self::assertCount(3, $results);
         self::assertSame([
-            'Operating System Concepts',
             'Database System Concepts',
             'Modern Operating Systems',
-        ], $results->all());
+            'Operating System Concepts',
+        ], $results->sort()->values()->all());
+    }
+
+    public function testDatabaseBuilderVectorSearch()
+    {
+        $results = $this->getConnection('mongodb')->table('books')
+            ->vectorSearch(
+                index: 'vector',
+                path: 'vector4',
+                queryVector: $this->vectors[7], // This is an exact match of the vector
+                limit: 4,
+                exact: true,
+            );
+
+        self::assertInstanceOf(LaravelCollection::class, $results);
+        self::assertCount(4, $results);
+        self::assertSame('The Art of Computer Programming', $results->first()['title']);
+        self::assertSame(1.0, $results->first()['vectorSearchScore']);
+    }
+
+    public function testEloquentBuilderVectorSearch()
+    {
+        $results = Book::vectorSearch(
+            index: 'vector',
+            path: 'vector4',
+            queryVector: $this->vectors[7],
+            limit: 5,
+            numCandidates: 15,
+            // excludes the exact match
+            filter: Query::query(
+                title: Query::ne('The Art of Computer Programming'),
+            ),
+        );
+
+        self::assertInstanceOf(EloquentCollection::class, $results);
+        self::assertCount(5, $results);
+        self::assertInstanceOf(Book::class, $results->first());
+        self::assertNotSame('The Art of Computer Programming', $results->first()->title);
+        self::assertSame('The Mythical Man-Month: Essays on Software Engineering', $results->first()->title);
+        self::assertThat(
+            $results->first()->vectorSearchScore,
+            self::logicalAnd(self::isType('float'), self::greaterThan(0.9), self::lessThan(1.0)),
+        );
+    }
+
+    /** Generate random vectors */
```

Comment:
I think this comment would benefit from noting that you're using a fixed seed to make the tests deterministic.

Also, am I correct in assuming the `vector4` generation her pertains to [How to Ingest Pre-Quantized Vectors](https://www.mongodb.com/docs/atlas/atlas-vector-search/vector-quantization/#how-to-ingest-pre-quantized-vectors) in the Atlas docs? That may be worth referencing as well.